integration_tests/src/main/python/string_test.py

```diff
@@ -561,6 +561,28 @@ def test_re_replace_backrefs():
         ),
         conf=_regexp_conf)
 
+def test_re_replace_anchors():
+    gen = mk_str_gen('.{0,5}TEST[\ud720 A]{0,5}TEST[\r\n\u0085\u2028\u2029]?')
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: unary_op_df(spark, gen).selectExpr(
+            'REGEXP_REPLACE(a, "TEST$", "")',
+            'REGEXP_REPLACE(a, "TEST$", "PROD")',
+            'REGEXP_REPLACE(a, "\ud720[A-Z]+$", "PROD")',
+            'REGEXP_REPLACE(a, "(\ud720[A-Z]+)$", "PROD")',
+            'REGEXP_REPLACE(a, "(TEST)$", "$1")',
+            'REGEXP_REPLACE(a, "^(TEST)$", "$1")',
+            'REGEXP_REPLACE(a, "\\\\ATEST\\\\Z", "PROD")',
```

Comment:
There are a huge number of possible patterns for the string gen, but only a very few of them would match this, and many of the patterns below it. Can we confirm that positive cases show up in the input data, restrict the `.` at the beginning to be smaller so there are less choices, or add in some special cases to make sure that things like "TEST\n" show up in the output?

Reply:
updated integration test in latest commit